### DISCUSSION TITLE
OCR Subtitles

### DISCUSSION THREAD

**Discussion by chenlung** [URUWorks/TeroSubtitler#12] [Ideas]
Is [Tesseract](https://github.com/tesseract-ocr/tesseract) the best engine for performing optical character recognition on subtitles? Should Tero Subtitler include such functionality?

My experience with soft-coded subtitles:

- [SubRip](https://en.wikipedia.org/wiki/SubRip): hasn't been updated for years, only appears to output SRT, and is Windows-only.
- [Subler](https://subler.org): outputs SRT and is macOS-only.

My experience with hard-coded subtitles:
- [Video-subtitle-extractor](https://github.com/YaoFANGUK/video-subtitle-extractor)

**Comment by GitGianluc** [comment]
@URUWorks how did you get a file like [TeroSubtitler/bin/OCR/Spanish.ocr](https://github.com/URUWorks/TeroSubtitler/blob/main/TeroSubtitler/bin/OCR/Spanish.ocr)? Did you start up by the relevant Tesseract tessdata file [tessdata / spa.traineddata](https://github.com/tesseract-ocr/tessdata/blob/main/spa.traineddata)?
I was wondering if and how it is possible to create the .ocr file for the Italian language as well.

**Reply by URUWorks** [reply]
it was done by hand in early SW versions :)

**Reply by GitGianluc** [reply]
Wow, I’m not sure if I am capable of producing something similar for the Italian language! I’ll think it over… 🤔

**Comment by URUWorks** [comment]
everything is possible

**Reply by GitGianluc** [reply]
Yeah, but I need further information: what is the difference between `"Method" : "WORD"`, `"UseRE" : "False"` and none of the previous? In other words, what do those options mean?

**Comment by URUWorks** [comment]
Find: Text to find
ReplaceBy: New text

Method:
WORD: The text is a word
LINE: Find the text on the line
BEGIN: The line begins with the text
END: The line ends with the text

UseRE:
True: Use regular expression
False: Do not use regular expression

**Reply by GitGianluc** [reply]
Thanks! 😊
